### PR TITLE
chore: removes use for declarations map and uses get binding on the parent path scope instead

### PR DESCRIPTION
**Pull request by madou**

### CODE REVIEW

File: packages/babel-plugin/src/utils/ast.tsx [hunk 1]
```diff
@@ -113,29 +113,28 @@ export const getKey = (node: t.Expression) => {
...
   let value: t.Node | undefined | null = undefined;
 
   if (t.isIdentifier(expression)) {
-    const binding = state.declarations[expression.name];
-
-    if (t.isVariableDeclaration(binding) && binding.kind === 'const') {
-      value = binding.declarations[0].init;
+    const binding = meta.parentPath.scope.getBinding(expression.name);
+    if (binding && binding.kind === 'const' && t.isVariableDeclarator(binding.path.node)) {
+      value = binding.path.node.init;
```
Comment (pgmanutd) [inline packages/babel-plugin/src/utils/ast.tsx:125]: ah!! finally no more index access `binding.declarations[0]`. Binding gives us `VariableDeclarator` directly 🎉

File: packages/babel-plugin/src/utils/ast.tsx [hunk 2]
```diff
@@ -113,29 +113,28 @@ export const getKey = (node: t.Expression) => {
...
     const { accessPath, bindingIdentifier } = getMemberExpressionMeta(expression);
-    const binding = state.declarations[bindingIdentifier.name];
+    const binding = meta.parentPath.scope.getBinding(bindingIdentifier.name);
 
     if (
-      t.isVariableDeclaration(binding) &&
+      binding &&
       binding.kind === 'const' &&
-      t.isObjectExpression(binding.declarations[0].init)
+      t.isVariableDeclarator(binding.path.node) &&
```
Comment (pgmanutd) [inline packages/babel-plugin/src/utils/ast.tsx:134]: nit: should we extract this into a function? 🤔 We have used it twice atleast

```typescript
const isBindedVariableDeclarator = (binding, kind = 'const') => binding && binding.kind === kind && t.isVariableDeclarator(binding.path.node);
```

and probably for this too `binding.path.node.init`

**Comment by pgmanutd** [comment]
Damn!! loved these changes ❤️. Finally we are getting rid of `state.declarations`. Less states means less bugs 😛

**Comment by pgmanutd** [comment]
Looks good. Go ahead with merging @madou. Just check one review comment I left. If you think it's not important atm, no worries then.

**Review by pgmanutd — APPROVED** [review]

File: packages/babel-plugin/src/utils/ast.tsx (hunk 2 above)
Comment (madou) [reply packages/babel-plugin/src/utils/ast.tsx:134]: good idea

**Comment by madou** [comment]
thanks! will merge - just rebased from master

good job with your examples!!

**Pull request merged**